src/profile/external-identities/FrameworkServiceLink.tsx

```diff
@@ -0,0 +1,37 @@
+import { ClickableProps } from '@terra-ui-packages/components';
+import React, { ReactNode } from 'react';
+import { ButtonPrimary, Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import { withErrorReporting } from 'src/libs/error';
+import * as Utils from 'src/libs/utils';
+
+interface FrameworkServiceLinkProps extends ClickableProps {
+  linkText: string;
+  providerKey: string;
+  redirectUrl: string;
+  button?: boolean;
+}
+
+export const FrameworkServiceLink = (props: FrameworkServiceLinkProps): ReactNode => {
+  const { linkText, providerKey, redirectUrl, button = false, ...clickableProps } = props;
+  const Component = button ? ButtonPrimary : Link;
+  const style = button ? {} : { display: 'inline-flex', alignItems: 'center' };
+  return (
+    <Component
+      onClick={async () => {
+        const loadAuthUrlAndRedirect = withErrorReporting('Error getting Fence Link', async () => {
```

Comment:
It looks like there are a couple of unnecessary wrapper functions here. I think this could all be done directly in the click handler.

```suggestion
      onClick={withErrorReporting('Error getting Fence Link', async () => {
```